drivers/counter/counter_mcux_gpt.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2019, Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <counter.h>
+#include <fsl_gpt.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(mcux_gpt, CONFIG_COUNTER_LOG_LEVEL);
+
+struct mcux_gpt_config {
+	/* info must be first element */
+	struct counter_config_info info;
+	GPT_Type *base;
+	clock_name_t clock_source;
+};
+
+struct mcux_gpt_data {
+	counter_alarm_callback_t alarm_callback;
+	counter_top_callback_t top_callback;
+	void *alarm_user_data;
+	void *top_user_data;
+};
+
+static int mcux_gpt_start(struct device *dev)
+{
+	const struct mcux_gpt_config *config = dev->config->config_info;
+
+	GPT_StartTimer(config->base);
+
+	return 0;
+}
+
+static int mcux_gpt_stop(struct device *dev)
+{
+	const struct mcux_gpt_config *config = dev->config->config_info;
+
+	GPT_StopTimer(config->base);
+
+	return 0;
+}
+
+static u32_t mcux_gpt_read(struct device *dev)
+{
+	const struct mcux_gpt_config *config = dev->config->config_info;
+
+	return GPT_GetCurrentTimerCount(config->base);
+}
+
+static int mcux_gpt_set_alarm(struct device *dev, u8_t chan_id,
+			      const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct mcux_gpt_config *config = dev->config->config_info;
+	struct mcux_gpt_data *data = dev->driver_data;
+
+	u32_t current = mcux_gpt_read(dev);
+	u32_t ticks = alarm_cfg->ticks;
+
+	if (chan_id != 0) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	if (!alarm_cfg->absolute) {
+		ticks += current;
+	}
+
+	data->alarm_callback = alarm_cfg->callback;
```

Comment:
it's missing checking if alarm is already in use:
```
if (data->alarm_callback) {
    return -EBUSY;
}
```

Reply:
done

Reply:
@nordic-krch could you please review again